core/internal/runhistoryreader/parquet/row_iterator.go

```diff
@@ -0,0 +1,746 @@
+package parquet
+
+// This file copies most of its logic from how history parquet files
+// are read on the backend.
+// See: https://github.com/wandb/core/blob/master/services/gorilla/internal/parquet/row_iterator.go
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"slices"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/parquet/file"
+	"github.com/apache/arrow-go/v18/parquet/pqarrow"
+	"github.com/apache/arrow-go/v18/parquet/schema"
+)
+
+const (
+	StepKey      = "_step"
+	TimestampKey = "_timestamp"
+)
+
+type HistoryPageParams struct {
+	MinStep int64 // Inclusive lower bound of _step
+	MaxStep int64 // Exclusive upper bound of _step
+	Samples int64 // Number of requested samples
+	MaxSize int64 // Maximum number of bytes we try to read (in some impls)
+}
+
+type EventsPageParams struct {
+	MinTimestamp float64
+	MaxTimestamp float64
+}
+
+// KVMap is a key-value pair mapping,
+// it represents a single metric for a history row.
+// Where Key is the metric name and Value is the metric value.
+type KVMap struct {
+	Key   string
+	Value any
+}
+
+// KVMapList is a list of KVMaps which represent a single history row.
+type KVMapList []KVMap
```

Comment:
`KVMap` is not a map; it's a key and value pair. How about the names `KeyValuePair` and `KeyValueList`?